pkgs/jni/example/lib/main.dart

```diff
@@ -37,21 +38,25 @@ int randomUsingEnv(int n) => using((arena) {
       final random = env.NewObject(randomCls, ctor);
       final nextInt = env.GetMethodID(randomCls, "nextInt".toNativeChars(arena),
           "(I)I".toNativeChars(arena));
-      final res = env.CallIntMethodA(random, nextInt, Jni.jvalues([n]));
-      env.deleteAllRefs([randomCls, random]);
+      final res =
+          env.CallIntMethodA(random, nextInt, toJValues([n], allocator: arena));
+      env.DeleteGlobalRef(randomCls);
+      env.DeleteGlobalRef(random);
       return res;
     });
 double randomDouble() {
-  final math = Jni.findJClass("java/lang/Math");
-  final random = math.callStaticMethodByName<double>("random", "()D", []);
+  final math = JClass.forName("java/lang/Math");
+  final random =
+      math.staticMethod("random", "()D").call(math, const jdoubleType(), []);
   math.release();
   return random;
 }
 
 int uptime() {
-  return Jni.findJClass("android/os/SystemClock").use(
-    (systemClock) => systemClock.callStaticMethodByName<int>(
-        "uptimeMillis", "()J", [], JniCallType.longType),
+  return JClass.forName("android/os/SystemClock").use(
+    (systemClock) => systemClock
+        .staticMethod("uptimeMillis", "()J")
+        .call(systemClock, const jlongType(), []),
```

Comment:
`JStaticMethod` should probably be a class so that `systemClock` can be saved in a field of it rather than having to repeat it in `.call`. Also see my comment on `JStaticMethod`.